STYLEGUIDE.md

```diff
@@ -0,0 +1,230 @@
+
+Each of the sections below would be the scope of a single readme:
+```
+ROOT README
+# Section (eg: api)
+ - packages
+```
+```
+# package
+ - classes
+```
+```
+# Class
+ - methods
+  - params
+```
+Each of these READMEs should only contain the table of contents of it's relevant data.
+___
+
+## C# Packages
+
+Each package directory should contain the following sections in its README:
+```md
+<Package overview>
+## Table of contents
+* [classes](_csharp-api/packagename/classes/)
+* [childclasses](_csharp-api/packagename/childclasses/)
+* ...
+## Classes
+* [classname](_csharp-api/packagename/classes/classname/)
+* ...
+## Child Classes
+* [classname](_csharp-api/packagename/childclasses/classname/)
+* ...
+## Template Classes
+* [classname](_csharp-api/packagename/templateclasses/classname/)
+* ...
+## Enums
+* [enumname](_csharp-api/packagename/enums/enumname/)
+* ...
+## Structs
+* [structname](_csharp-api/packagename/structs/structname/)
+* ...
+## Interfaces
+* [interfacename](_csharp-api/packagename/interfaces/interfacename/)
+* ...
+## Exception Classes
+* [exceptionname](_csharp-api/packagename/exceptions/exceptionname/)
+* ...
+```
+>Each of the sections (at both levels) should have a valid link to their respective info pages.
+
+The package information header should contain the following:
+```md
+Current Version (Stable branch): v0.0.0.0
+<Brief description of the package and its contents>
+```
+___
+
+## Classes (Base, Child, Templated, and Exceptions)
+
+Unlike the root and package level READMEs, the class README will contain more than just links to child folders. Each class will start with a **Class Overview Panel**:
+
+### [Parent Object Name](EXAMPLE)
+# Class Name
+
+This is a description of the class and what it does.
+
+<span style= "font-size:12px">
+
+Using / Namespaces | | <span style='color:yellow'>Taleworlds.Core</span>| | | | <span style='color:green'>Saveable</span> |
+:---|---|---|---|---|---|---:|
+[System](EXAMPLE) | | | | | | 10000 |
+[System.Xml](EXAMPLE)|
+[TaleWorlds.SaveSystem](EXAMPLE) |
+|
+</span>
+
+> Omit elements entirely if not relavent to that class (eg: parent class)
+
+> For the non-savable classes, use ```'color:red'``` and change to text to "Not Savable".
+
+After this class overview, a small code snippet should exist showing how to use / initialize the class. **This should not be exhaustive**, just a quick reference.
+```csharp
+new InformationMessage("Hello World!")
+```
+
+The above would be a valid snippet for the ```InformationMessage``` Class, but can be longer if the extra length is used to give important context about the snippet.
+
+Directly after this intro snippet, we have a section outlining the variables of the class. All variables are assumed to have private setters unless otherwise noted. Each variable should contain all of the following information:
+
+* Name
+* Type (with link to that object or objects)
+* Notes
+
+## Variables
```

Comment:
Where to describe consts?
in this section, or another?

Reply:
fixed :)

Reply:
If in the same table - value required, I think.